logic.py

```diff
@@ -998,4 +1000,76 @@ def export_workflow_report(article_list, averages):
 
         all_rows.append(row)
 
-    return export_csv(all_rows)
\ No newline at end of file
+    return export_csv(all_rows)
+
+
+def get_metrics_start_end(request):
+    """
+    Should be used in combination with DateRangeForm. Fetches start_date and
+    end_date from request.GET and transforms them into datetimes better
+    suited for filtering accesses.
+    """
+    try:
+        try:
+            # Get start and end date from request.GET
+            start_date = datetime.strptime(
+                request.GET.get('start_date'),
+                '%Y-%m-%d',
+            )
+            end_date = datetime.strptime(
+                request.GET.get('end_date'),
+                '%Y-%m-%d',
+            ).replace(
+                hour=23,
+                minute=59,
+                second=59,
+            )
+        except ValueError:
+            start_date, end_date = None, None
+            messages.add_message(
+                request,
+                messages.WARNING,
+                'Date not in recognised format Y-m-d',
+            )
+    except TypeError:
+        # No defaults supplied at all, set to None and generate below.
+        start_date, end_date = None, None
+
+    if not start_date and not end_date:
+        # Calculate the start and end dates for this month
+        current_date = timezone.now()
+        start_date = current_date.replace(day=1)
+        end_date = start_date + relativedelta(
+            months=1,
+            days=-1,
+            hour=23,
+            minute=59,
+            second=59,
+        )
+    return start_date, end_date
+
+
+def manager_metrics_summary(repository, start_date, end_date):
+    preprints = repository_models.Preprint.objects.filter(
+        repository=repository,
+        preprintaccess__accessed__gte=start_date,
+        preprintaccess__accessed__lte=end_date
+    ).annotate(
+        total_views=Count(
+            'preprintaccess',
+            filter=Q(
+                preprintaccess__file=None,
+                preprintaccess__accessed__gte=start_date,
+                preprintaccess__accessed__lte=end_date,
```

Comment:
users are asked to provide "Y-m-d" while our default provides `"Y-m-d:HH-mm-dd"`. I think this will lead to some minor differences between the default range and a user manually entering the range (The "Y-m-d" dates will have their time portion set to 00:00:00 automatically). Since some users seem to be picky about this, shall we filter just using the __date filter in all cases?

https://docs.djangoproject.com/en/3.2/ref/models/querysets/#date

Reply:
Yes - tested this and it works.